training/trillium/Llama3.1-70B-MaxText-with-Storage/README.md

```diff
@@ -0,0 +1,189 @@
+- `<REGION>`: the region where your GKE cluster is located ([available locations](https://cloud.google.com/storage/docs/locations#location-r))
+
+2. Follow these [instructions](https://github.com/AI-Hypercomputer/maxtext/blob/b93beba652db6b3f4e6c82dc48a83b03229f5d3a/getting_started/Data_Input_Pipeline.md#tfds-pipeline) to download the Allenai c4 dataset to the dataset bucket.
+Then follow these [instructions](https://github.com/google/array_record/tree/main/beam) to convert the dataset into ArrayRecord.
+
+## XPK setup
+1. Please follow this [link](https://github.com/AI-Hypercomputer/tpu-recipes/blob/main/training/trillium/XPK_README.md) to create your GKE cluster with XPK.
+2. GCSFuse lets you mount and access Cloud Storage buckets as local file systems, so applications can read and write objects in your bucket using standard file system semantics. You'll need to use the below commands to create [XPK storage resources](https://github.com/AI-Hypercomputer/xpk?tab=readme-ov-file#storage) for both the dataset and checkpoint buckets in order to mount them to the MaxText workload using GCSFuse. For the dataset bucket and checkpoint bucket use separate manifest files `checkpoint_pvc.yaml` and `dataset_pvc.yaml` from this repo.
+Be sure to update `volumeHandle` in the yamls with your correct bucket names. Creating a bucket and xpk storage is a one time setup.
+```
+export RECIPE_REPO="path-to-this-recipe-repo" # Update
+
+cd ~/xpk
+
+python3 xpk.py storage attach dataset-bucket type=gcsfuse project=$PROJECT cluster=$CLUSTER zone=$ZONE mountpoint=/tmp/dataset readonly=false bucket=$DATASET_BUCKET size=64 automount=false manifest=$RECIPE_REPO/tpu-recipes/training/trillium/Llama3.1-70B-MaxText-with-Storage/dataset_pvc.yaml
+
+python3 xpk.py storage attach checkpoint-bucket type=gcsfuse project=$PROJECT cluster=$CLUSTER zone=$ZONE mountpoint=/tmp/ckpt readonly=false bucket=$CHECKPOINT_BUCKET size=64 automount=false manifest=$RECIPE_REPO/tpu-recipes/training/trillium/Llama3.1-70B-MaxText-with-Storage/checkpoint_pvc.yaml
+```
+
+
+## Prep for MaxText
+
+### Install MaxText and Build Docker Image
+Please follow this [link](https://github.com/AI-Hypercomputer/tpu-recipes/blob/main/training/trillium/MAXTEXT_README.md) to install maxtext and build the docker image.
+
+In step 2, use the jax-stable-stack image containing JAX 0.5.2:
+```
+BASE_IMAGE=us-docker.pkg.dev/cloud-tpu-images/jax-stable-stack/tpu:jax0.5.2-rev1
+bash docker_build_dependency_image.sh DEVICE=tpu MODE=stable_stack BASEIMAGE=${BASE_IMAGE}
+```
+
+## Run MaxText Llama3.1-70B workloads on GKE
+
+### Starting workload
+
+From the MaxText root directory, start your Llama3.1-70B workload.
+
+Run MaxText Llama 3.1 70B with synthetic data and no checkpointing:
+```
+python3 benchmarks/benchmark_runner.py xpk \
+    project=$PROJECT \
+    zone=$ZONE \
+    device_type=v6e-256 \
+    num_slices=1  \
+    cluster_name=$CLUSTER \
+    base_output_directory=$OUTPUT_DIR \
```

Comment:
I just noticed that there are some env variables here that users are not instructed to populate above.